sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -21,13 +21,66 @@
 service.
 """
 
+from collections import defaultdict
+
+from apache_beam.metrics.execution import MetricKey
+from apache_beam.metrics.execution import MetricResult
 from apache_beam.metrics.metric import MetricResults
+from apache_beam.metrics.metricbase import MetricName
 
 
 # TODO(pabloem)(JIRA-1381) Implement this once metrics are queriable from
 # dataflow service
 class DataflowMetrics(MetricResults):
+  """Implementation of MetricResults class for the Dataflow runner."""
+
+  def __init__(self, dataflow_client=None, job_result=None):
+    super(DataflowMetrics, self).__init__()
+    self._dataflow_client = dataflow_client
+    self.job_result = job_result
+
+  def _populate_metric_results(self, response):
+    """Take a list of metrics, and convert it to a list of MetricResult."""
+    user_metrics = [metric
+                    for metric in response.metrics
+                    if metric.name.origin == 'user']
+
+    # Get the tentative/committed versions of every metric together.
+    metrics_by_name = defaultdict(lambda: {})
+    for metric in user_metrics:
+      tentative = [prop
+                   for prop in metric.name.context.additionalProperties
+                   if prop.key == 'tentative' and prop.value == 'true']
+      key = 'tentative' if tentative else 'committed'
+      metrics_by_name[metric.name.name][key] = metric
+
+    # Now we create the metricresult elements
+    result = []
+    for name, metric in metrics_by_name.iteritems():
+      if (name.endswith('(DIST)') or
+          name.endswith('[MIN]') or
```

Comment:
The distribution metrics are a bit 'hacked' ATM. We are not setting suffixes for non-distribution metrics, and we will change the naming soon, once structured names are completely wired in. 
As a note, since we're not adding counter suffix, it is actually possible to create a counter which name overlaps with a distribution, but we decided to ignore this possibility.
Also, yes, the mismatch is intentional.